### ISSUE TITLE
Feature request: Toggle for StrictUndefined behavior

### ISSUE THREAD

**Issue by davidfischer-ch** [NikolaLohinski/terraform-provider-jinja#3]
Hello,

Thank you for this nice provider !

I am almost always looking for a templating engine to raise an error in case a variable is undefined. Unfortunately I think I cannot define this behavior through the provider.

https://stackoverflow.com/questions/17434893/raise-an-exception-for-undefined-attributes-in-jinja2

It can be an idea to add parameter both in the provider (to define default value) and in the data source (to set this option individually).

What do you think of this feature?

Thank you.

**Comment by NikolaLohinski** [comment]
Hi and thank you for your feedback, I really appreciate it :pray: 

I would definitely like to have this feature somehow embeded in ths provider. However, from the looks of it, it's not something that is available in the underlying jinja engine (gonja).

I am already using a fork of the gonja lib in this provider because the original repo is unmaintained. Therefore PRs are definitely welcome in https://github.com/NikolaLohinski/gonja if you feel up to it.

The only half solution I can suggest with the current implementation is for you to leverage the [JSON schema validation](https://registry.terraform.io/providers/NikolaLohinski/jinja/latest/docs/data-sources/template#schema) that happens prior to rendering the template. That definitely does not cover all use-cases, but it's better than no validation whastoever.

**Comment by NikolaLohinski** [comment]
Hi @davidfischer-ch .

I implemented this toggle (via `strict_undefined` at either the provider or the data source level) in [my fork of gonja](https://github.com/NikolaLohinski/gonja) and added it in [the latest `v1.4.0` release](https://github.com/NikolaLohinski/terraform-provider-jinja/releases/tag/v1.4.0). Let me know if that fits your needs and please do report any issues. Otherwise feel free to leave me a star on both repos :smiley:  

Cheers

**Comment by davidfischer-ch** [comment]
Hi @NikolaLohinski,

Thank you. Unfortunately it seems that the v1.4.0 accepts the `strict_undefined` boolean but it is not changing the behavior on templates with undefined variables (they are replaced by "emptiness" either way).

```
$ terraform version
Terraform v1.3.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/local v2.2.3
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/kreuzwerker/docker v2.23.0
+ provider registry.terraform.io/nikolalohinski/jinja v1.4.0

Your version of Terraform is out of date! The latest version
is 1.3.6. You can update by downloading from https://www.terraform.io/downloads.html
```

```
terraform {
  required_version = ">= 1.3"

  required_providers {
   (...)
    jinja = {
      # 18-12-2022 Released 16-12-2022
      source  = "NikolaLohinski/jinja"
      version = "1.4.0"
    }
   (...)
  }
}

provider "jinja" {
  strict_undefined = true
}

data "jinja_template" "sites_config" {
  for_each = var.sites

  template = "${path.module}/config/sites/${each.value.name}.conf.j2"
  context {
    type = "json"
    data = each.value
  }
}

```

Content of one of the sites `.j2` :

```
{{ salut }}
```

Do you need more details?

**Comment by NikolaLohinski** [comment]
Hi @davidfischer-ch 

Great catch, thanks a lot for it, and for your time looking into it.

It should be fixed in v1.4.1

Let me know if you run into any issues, or feel free to close the issue :)

**Comment by davidfischer-ch** [comment]
Awesome its working properly !

Thank you for your work.
